wdl2cwl/tests/wdl_files/common.wdl

```diff
@@ -0,0 +1,297 @@
+version 1.0
+
+# Copyright (c) 2017 Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task AppendToStringArray {
+    input {
+        Array[String] array
+        String string
+    }
+
+    command {
+        echo "~{sep='\n' array}
+        ~{string}"
+    }
+
+    output {
+        Array[String] outArray = read_lines(stdout())
+    }
+
+    runtime {
+        memory: "1G"
+    }
+}
+
+# This task will fail if the MD5sum doesn't match the file.
+task CheckFileMD5 {
+    input {
+        File file
+        String md5
+        # By default cromwell expects /bin/bash to be present in the container.
+        # The 'bash' container does not fill this requirement. (It is in /usr/local/bin/bash)
+        # Use a stable version of debian:stretch-slim for this. (Smaller than ubuntu)
+        String dockerImage = "debian@sha256:f05c05a218b7a4a5fe979045b1c8e2a9ec3524e5611ebfdd0ef5b8040f9008fa"
+    }
+
+    command {
+        bash -c '
+        set -e -o pipefail
+        echo "~{md5}  ~{file}" | md5sum -c
+        '
+    }
+
+    runtime {
+        docker: dockerImage
```

Comment:
True, `dockerPull` can not be a dynamic string as of CWL v1.2. In this case it would be good to detect that `dockerImage` was an input with a default value and pull that static string value over. (And issue a warning).